msdsl/generator/verilog.py

```diff
@@ -382,15 +382,15 @@ def make_constant_mul_signal(self, expr: Product):
         # create the output signal
         output = Signal(name=next(self.namer), format_=expr.format_)
 
-        # call the special multiplication macro, avoiding a synthesis corner-case:
-        # https://forums.xilinx.com/t5/Synthesis/Possible-synthesis-bug-casting-integer-to-real-in-a-function/td-p/1140910
-        if isinstance(constant.value, Integral):
+        # if multiplying by -1, use a special macro to reduce utilization
+        if constant.value == -1:
+            self.macro_call('NEGATE_REAL', signal.name, output.name)
```

Comment:
When multiplying -1 by a variable, use the special NEGATE_REAL macro to avoid consuming a DSP slice.  I thought the synthesis tool would do that anyway, but it only seems to do that optimization sometimes.